src/bijectors/planar_layer.jl

```diff
@@ -15,73 +15,93 @@ using Roots # for inverse
 
 # TODO: add docstring
 
-# FIXME: using `TrackedArray` for the parameters, we end up with
-# nested tracked structures; don't want this.
-mutable struct PlanarLayer{T1,T2} <: Bijector{1}
+mutable struct PlanarLayer{T1 <: AbstractVector{<:Real}, T2 <: Real} <: Bijector{1}
     w::T1
     u::T1
     b::T2
 end
 
 function get_u_hat(u, w)
     # To preserve invertibility
-    # From A.1
-    return u .+ (planar_flow_m.(w' * u) .- w' * u) .* w ./ sum(abs2, w)
+    x = w' * u
+    return u .+ (planar_flow_m(x) - x) .* w ./ sum(abs2, w)   # from A.1
 end
 
 function PlanarLayer(dims::Int, container=Array)
-    w = container(randn(dims, 1))
-    u = container(randn(dims, 1))
-    b = container(randn(1))
+    w = container(randn(dims))
+    u = container(randn(dims))
+    b = randn()
```

Comment:
I think this came up in the discussion in some other PR, but I believe the reasoning for making `b` an array is so that in the case were one is using Tracker, one can use `TrackedArray` and update `b` by doing elementwise assignment. When this is a `Real` we have to reconstruct the `PlanarLayer`. 

It's kind of ugly to have it as a one-element array, but at the same time I understand the original reasoning as it's quite convenient. I'm not certain what I prefer, personally.

Reply:
You cannot update a `TrackedArray` in-place. `Array{TrackedReal}` yes. I think reconstructing the whole thing is reasonable. No need to use a 1 element array. If we really must make it mutable, we can use a `Ref`. A `Ref` is lighter on the compiler than an array.

Reply:
This issue just came up: https://github.com/TuringLang/Bijectors.jl/issues/121, and I remembered that we had a discussion exactly about this part of the code but couldn't remember the conclusion. It also seemed weird that I had "approved" such a change because it breaks with the users expectations. So it seems that the fact that it was marked as resolved made it go under my radar :confused:

So for future reference: I think it's a good idea to let the "asker" resolve the question rather than the "answerer" :+1:

Reply:
Also, just to clarify: I think `b` should still be wrapped in `container` and change the default `container` to just be `identity` (since `randn` returns an `Array` anyways) or we add another argument specific for the `Real` parameter. It's useful when you're using, say, `Tracker.param`.

Reply:
Maybe just rename the argument to `wrapper` or something

Reply:
Sounds good, sorry about that!

Reply:
No worries; might also have been my fault in the sense that I wasn't quick enough to address the comment after you answered :+1: